src/init.cpp

```diff
@@ -536,22 +536,24 @@ class BlockNotifyCaller : public CValidationInterface {
         }
     }
 } g_blocknotify_caller;
-} // anonymous namespace
 
 static bool fHaveGenesis = false;
```

Comment:
I'd rather see more of this encapsulated in the GenesisWaiter. `AppInitMain` could be simplified if `GenesisWaiter` just had a method like `BlockUntilHaveGenesis` or something.

Reply:
Done.